astropy/coordinates/tests/test_velocity_corrs.py

```diff
@@ -0,0 +1,199 @@
+    2457244.50268     0.00   -14.45     5.44      0.146    0.001  -14.600   19.897
+    2457244.50446     0.00     3.84    19.33      0.356    0.008    3.478   15.491
+    2457244.50325     0.00    21.97    26.39      0.357    0.011   21.598    4.419
+    2457244.49975     0.00    29.30    22.47      0.149    0.008   29.146   -6.831
+    2457244.49602     0.00    21.55     9.88     -0.146   -0.001   21.700  -11.670
+    2457244.49423     0.00     3.26    -4.00     -0.356   -0.009    3.623   -7.263
+    2457244.49544     0.00   -14.87   -11.06     -0.357   -0.011  -14.497    3.808
+    2457244.49561     0.00   -11.13    -5.46     -0.315   -0.010  -10.805    5.670
+    2457244.49921     0.00   -17.43    -0.77     -0.096   -0.006  -17.333   16.664
+    2457244.50269     0.00    -6.75    12.83      0.196    0.003   -6.949   19.583
+    2457244.50344     0.00    12.88    25.10      0.340    0.009   12.527   12.227
+    2457244.50089     0.00    26.67    26.80      0.228    0.009   26.430    0.137
+    2457244.49696     0.00    24.24    16.65     -0.056    0.002   24.290   -7.584
+    2457244.49461     0.00     7.42     2.29     -0.297   -0.007    7.719   -5.122
+    2457244.49561     0.00   -11.13    -5.46     -0.315   -0.010  -10.805    5.670
+    2457244.49598     0.00    -6.90     0.38     -0.259   -0.008   -6.629    7.277
+    2457244.49949     0.00   -11.53     5.78     -0.038   -0.004  -11.489   17.311
+    2457244.50232     0.00     1.84    19.37      0.221    0.004    1.612   17.533
+    2457244.50165     0.00    19.84    27.56      0.259    0.008   19.573    7.721
+    2457244.49814     0.00    24.47    22.16      0.038    0.004   24.433   -2.313
+    2457244.49530     0.00    11.11     8.57     -0.221   -0.005   11.332   -2.534
+    2457244.49598     0.00    -6.90     0.38     -0.259   -0.008   -6.629    7.277
+    2457244.49652     0.00    -2.35     6.21     -0.192   -0.006   -2.156    8.560
+    2457244.50041     0.00    -2.13    15.06      0.078    0.000   -2.211   17.194
+    2457244.50071     0.00    17.41    26.30      0.192    0.006   17.214    8.888
+    2457244.49683     0.00    17.19    17.44     -0.078   -0.001   17.269    0.253
+    2457244.49652     0.00    -2.35     6.21     -0.192   -0.006   -2.156    8.560
+    2457244.49722     0.00     2.29    11.75     -0.115   -0.004    2.413    9.459
+    2457244.49975     0.00    14.20    23.86      0.115    0.004   14.085    9.656
+    2457244.49722     0.00     2.29    11.75     -0.115   -0.004    2.413    9.459
+    2457244.49805     0.00     6.84    16.77     -0.034   -0.001    6.874    9.935
+    """
+    vhs_iraf = []
+    for line in rvcorr_result.strip().split('\n'):
+        if not line.strip().startswith('#'):
+            vhs_iraf.append(float(line.split()[2]))
+    vhs_iraf = vhs_iraf*u.km/u.s
+
+    vhs_astropy = radial_velocity_correction(IRAF_time, loc, coos)
+    assert_quantity_allclose(vhs_astropy, vhs_iraf, atol=150*u.m/u.s)
```

Comment:
I'd say IRAF's.  I compared the result of the barycentric correction to BARYCORR and it is right to 1-2 m/s (which I think you noted above).

My memory is a bit hazy now, but as I recall the IRAF algorithm is a bit ambiguous about how it defines some of its definitions relative to the modern way (or at least *I* didn't understand it). I remember thinking at the time, at least, that this didn't surprise me because of of imperfect IRAF-something. But at the *time* I didn't have a better one.  Of course BARYCORR counts as that now.